core/src/main/java/org/apache/calcite/rel/rules/AggregateValuesRule.java

```diff
@@ -75,10 +91,18 @@ public AggregateValuesRule(RelBuilderFactory relBuilderFactory) {
   @Override public void onMatch(RelOptRuleCall call) {
     final Aggregate aggregate = call.rel(0);
     final Values values = call.rel(1);
-    Util.discard(values);
     final RelBuilder relBuilder = call.builder();
     final RexBuilder rexBuilder = relBuilder.getRexBuilder();
 
+    if ((aggregate.getGroupCount() != 0 || !values.getTuples().isEmpty())
+        && aggregate.groupSets.size() == 1) {
+      List<ImmutableList<RexLiteral>> distinctValues =
+          values.getTuples().stream().distinct().collect(Collectors.toList());
+      relBuilder.values(distinctValues, values.getRowType());
```

Comment:
doesn't the field you group on need to be the same that is aggregated as well?
where is that check?
VALUES can contain ROW objects.

Reply:
Here would only handle with groupSets.size = 1 condition，I had a tried seems ROW type can not constructor in test case. Can we just handle with Literal type first？

Reply:
Mihai may mean this: 
```
    final Function<RelBuilder, RelNode> relFn = b -> b
        .values(new String[]{"v1", "v2"},  1, "a", 2, "b", 1, "a", 1, "b")
        .aggregate(b.groupKey(ImmutableBitSet.of(0)))
        .build();
```
This will generate the following plan, which only does a GROUP BY on the v1 field.
```
LogicalAggregate(group=[{0}])
  LogicalValues(tuples=[[{ 1, 'a' }, { 2, 'b' }, { 1, 'a' }, { 1, 'b' }]])
```
Maybe you need to check whether this scenario is supported.

Reply:
Had test for Row type about it and it would distinct result as expected.
```
  <TestCase name="testReduceInValuesWithAggregateValueReduceRuleWithRow">
    <Resource name="planBefore">
      <![CDATA[
LogicalAggregate(group=[{0, 1}])
  LogicalValues(tuples=[[{ 1, 'a' }, { 2, 'b' }, { 1, 'a' }, { 1, 'b' }, { 1, 'a' }]])
]]>
    </Resource>
    <Resource name="planAfter">
      <![CDATA[
LogicalValues(tuples=[[{ 1, 'a' }, { 2, 'b' }, { 1, 'b' }]])
]]>
    </Resource>
```

Reply:
You should use `aggregate(b.groupKey(ImmutableBitSet.of(0)))` to test. `distinct()` can not expose any problems.

Reply:
```
doesn't the field you group on need to be the same that is aggregated as well?
where is that check?
VALUES can contain ROW objects.
```
@mihaibudiu had added check for rowType. Need to be same with that is aggregated.